examples/experimental_component/README.md

```diff
@@ -0,0 +1,16 @@
+## Experimental component API example
+
+Demonstrates a component API that has:
+
+- Labeled required properties, ensured to be present at compile-time
+- Labeled optional properties
+- Polymorphic for all properties
+- A convenient consumer interface, through a stupidly simple macro that is very easy to understand
```

Comment:
We shouldn't assume all people understand the same things the same way and that they are all experienced enough to understand it on the first glance. It's not recommended to say that something is "very simple" or similar stuff in documentation / tutorials in general. (I'm sure I've written something similar too in my examples/ tutorials / docs but it shouldn't be there).
So please change it to something like:
```suggestion
- A convenient consumer interface, through a simpler macro
```
OT: Please update also CHANGELOG and I think I can merge it :)

Reply:
Ah, sorry, I didn't mean for this to be merged as-is, or for this text to be consumed by end-users just yet, but instead to facilitate discussion among contributors.

Before merging I'd like to move the macro into seed proper, behind an experimental feature flag, and perhaps add a `Component` trait so that it can't be used directly without setting up the (eventual) boundary, But as that would obscure the API a bit I wanted to do a round with just this first, to focus on the API only.

Since there doesn't seem to be any immediate concerns, however, I can do that later today and also try to fill in a bit more of the bigger picture.